server/apps/archive/archive_crop.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import superdesk
+from flask import request
+from superdesk import get_resource_service
+from superdesk.resource import Resource
+from .common import item_url
+from superdesk.services import BaseService
+from superdesk.errors import SuperdeskApiError
+from superdesk.media.media_operations import crop_image, process_file_from_stream
+from superdesk.upload import UploadService, url_for_media
+
+
+class ArchiveCropResource(Resource):
+    endpoint_name = 'archive_crop'
+    url = 'archive/<{0}:item_id>/crop/<{0}:crop_name>'.format(item_url)
+    schema = {
+        'CropLeft': {'type': 'integer'},
+        'CropRight': {'type': 'integer'},
+        'CropTop': {'type': 'integer'},
+        'CropBottom': {'type': 'integer'}
+    }
+    datasource = {'source': 'archive'}
+    resource_methods = ['GET', 'POST', 'DELETE']
+    resource_title = endpoint_name
+    privileges = {'POST': 'archive', 'DELETE': 'archive'}
+
+
+class ArchiveCropService(BaseService):
+
+    def create(self, docs, **kwargs):
+        item_id = request.view_args['item_id']
+        crop_name = request.view_args['crop_name'].lower()
+        archive_service = get_resource_service('archive')
+        original = archive_service.find_one(req=None, _id=item_id)
+        self._validate_crop(original, crop_name, docs[0])
+        self._add_crop(original, crop_name, docs[0])
+        return [original]
+
+    def delete(self, lookup):
+        item_id = request.view_args['item_id']
+        crop_name = request.view_args['crop_name'].lower()
+        archive_service = get_resource_service('archive')
+        original = archive_service.find_one(req=None, _id=item_id)
+        self._delete_crop(original, crop_name)
+
+    def _validate_crop(self, original, crop_name, doc):
+        # Check if type is picture
+        if original.get('type') != 'picture':
+            raise SuperdeskApiError.badRequestError(message='Only images can be cropped!')
+
+        # Check if the renditions exists
+        if not original.get('renditions'):
+            raise SuperdeskApiError.badRequestError(message='Missing renditions!')
+
+        # Check if the original rendition exists
+        if not original.get('renditions').get('original'):
+            raise SuperdeskApiError.badRequestError(message='Missing original rendition!')
+
+        # Check if the crop name is valid
+        crops = get_resource_service('vocabularies').find_one(req=None, _id='crop_sizes')['items']
```

Comment:
It's ok if we don't want to inform API Client on missing vocabulary. But if vocabulary isn't available that line will throw `KeyError`